Limit changing of record type. #638

knirirr opened the pull request:
This change disables the record type/registry field unless:

1. Creating a new record, or
2. Logged in as a curator.

The curators will want users to always be able to change registry/type later but that will involve more complex operations with the stored data and so is best left until after MVP (users can't change types in the current system anyway). 

This can be closed afterwards: https://github.com/FAIRsharing/FAIRsharing-API/issues/215

knirirr committed:
Limit changing of record type. #638

knirirr committed:
Autogenerated documentation files

terazus commented on src/components/Editor/GeneralInformation/BaseFields.vue:
```diff
@@ -364,16 +365,25 @@
+              return false;
+            } else {
+              return !_module.user().is_curator;
             }
```
else is not required:
```suggestion
            if (_module.getCreatingNewRecord) {
              return false;
            } 
            return !_module.user().is_curator;
```

terazus commented on src/store/record.js:
```diff
@@ -94,6 +95,9 @@ let recordStore = {
         resetRegistry(state){
             state.sections.generalInformation.data.type = "";
         },
+        setCreatingNewRecord(state){
```
Don't you also need a method to set it back to false when users are editing a record after creating a new record ?

knirirr replied:
Good point, I'll have to look into that.

knirirr committed:
Update src/components/Editor/GeneralInformation/BaseFields.vue

Looks good to me.

Co-authored-by: Dominique <batistadominique@hotmail.com>

knirirr committed:
Autogenerated documentation files

knirirr committed:
Changes as requested
https://github.com/FAIRsharing/fairsharing.github.io/pull/644#pullrequestreview-565536827

knirirr committed:
Autogenerated documentation files

knirirr committed:
Merge branch 'dev' into mt_disable_type_change_638

knirirr committed:
Merge branch 'mt_disable_type_change_638' of github.com:FAIRsharing/fairsharing.github.io into mt_disable_type_change_638

knirirr committed:
Autogenerated documentation files

knirirr committed:
Doc. removal for merging.

knirirr committed:
Merge branch 'dev' into mt_disable_type_change_638

knirirr committed:
Autogenerated documentation files
